src/devices/src/virtio/net/device.rs

```diff
@@ -409,24 +425,28 @@ impl Net {
     fn write_to_mmds_or_tap(
         mmds_ns: Option<&mut MmdsNetworkStack>,
         rate_limiter: &mut RateLimiter,
-        frame_buf: &[u8],
+        frame_buf: &mut [u8],
+        frame_iovec: &mut IoVecBuffer,
```

Comment:
I think you could get away with non-mutable `IoVecBuffer`; I can't think of a reason this should be mutable...
```suggestion
        frame_iovec: &IoVecBuffer,
```